capsule/src/main/java/Capsule.java

```diff
@@ -4368,19 +4368,27 @@ private static boolean equals(int[] a, int[] b, int n) {
         return true;
     }
 
-    private static final Pattern PAT_JAVA_VERSION = Pattern.compile("(?<major>\\d+)(\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?(_(?<update>\\d+))?(-(?<pre>[^-]+))?(-(?<build>.+))?");
+    private static final Pattern PAT_JAVA_VERSION = Pattern.compile("(?<major>\\d)(\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?(_(?<update>\\d+))?(-(?<pre>[^-]+))?(-(?<build>.+))?");
+    // from https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Runtime.Version.html
+    private static final Pattern PAT_JAVA_VERSION_JDK10 = Pattern.compile("(?<major>\\d+)(\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?(\\.(?<update>\\d+))?(-(?<pre>[^-]+))?.*");
```

Comment:
isn't it the contrary, update first then patch?


if I sum up, we have those string version formats:
- 1.major.minor_update from java 1.3 to 1.8
- major.minor.update from java 9
- major.minor.update.patch from java 10+

I think the java 9 format is way closer to java 10 format than java 1.3-8, except we will never have the patch part.
So we could have the first regex parsing "1.major.minor_update" up to java 8, and the second "major.minor.update.patch" from java 9 onward?

Reply:
Yes ok, done